Sources/Afterpay/BadgeView.swift

```diff
@@ -60,7 +68,23 @@ public final class BadgeView: UIView {
       svgView.topAnchor.constraint(equalTo: topAnchor),
       svgView.bottomAnchor.constraint(equalTo: bottomAnchor),
     ])
+  }
+
+  public override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+    super.traitCollectionDidChange(previousTraitCollection)
 
+    svgView.svg = style(for: traitCollection).svg
+  }
+
+  private func style(for traitCollection: UITraitCollection) -> Style {
+    switch traitCollection.userInterfaceStyle {
+    case .dark:
+      return darkStyle
+    case .light, .unspecified:
+      fallthrough
+    @unknown default:
+      return lightStyle
+    }
   }
```

Comment:
This might be nicer as an initialiser extension on `Style`.

Reply:
I started to give this a go, however it didn't pan out as expected. If I make it an initialiser then I'd need to store the preferred SVGs for light and dark if you see the issue. Trait collection is a thing that changes over time